packages/astro/src/@types/astro.ts

```diff
@@ -1487,6 +1513,31 @@ export interface APIContext<Props extends Record<string, any> = Record<string, a
 	 * [context reference](https://docs.astro.build/en/guides/api-reference/#contextredirect)
 	 */
 	redirect: AstroSharedContext['redirect'];
+
+	/**
+	 * Object accessed via Astro middleware.
+	 *
+	 * Example usage:
+	 *
+	 * ```ts
+	 * // src/middleware.ts
+	 * import {defineMiddleware} from "astro/middleware";
+	 *
+	 * export const onRequest = defineMiddleware((context, next) => {
+	 *   context.locals.greeting = "Hello!";
+	 *   next();
+	 * });
+	 * ```
+	 * Inside the an astro file:
```

Comment:
```suggestion
	 * Inside a `.astro` file:
```
We keep debating whether people say "a dot astro file" or "an astro file" even when written `.astro`. :sweat_smile:  We think most people pronounce the dot, so we tend to use this.